### PR TITLE
Misc code cleanup

### PR DESCRIPTION
**Pull request by hlinnaka**
Misc code cleanup, to reduce the diff with upstream, and hopefully make merging slightly easier.

### CODE REVIEW

**Comment by ashwinstar** [comment]
Looks good to me. @gcaragea @vraghavan78 please can you look at the spi related deletions specifically.

**Comment by asimrp** [comment]
How did you catch that the code using `_QD_conn` is dead?  Coverity should be smart enough to figure such things out and I am surprised it was not caught by our Coverity build.

File: src/backend/access/transam/xlog.c [hunk 1]
```diff
@@ -757,6 +759,7 @@ static char *str_time(pg_time_t tnow);
 #ifdef suppress
 static void issue_xlog_fsync(void);
 #endif
+static void xlog_outrec(StringInfo buf, XLogRecord *record);
```
Comment (asimrp) [inline src/backend/access/transam/xlog.c]: What's the convention we should follow?  Static functions should be named in lower case?

**Comment by hlinnaka** [comment]
On 11/06/2015 08:23 PM, Asim RP wrote:
> How did you catch that the code using `_QD_conn` is dead?

Old-fashioned "grep -r _QD_conn src/backend" :-).

The comment was the first clue though. "If we are connected back to the 
QD" seemed odd to me, because I thought that we don't support that. And 
indeed we don't, and it was dead code.

And the reason I looked at this piece of code in the first place is that 
I was looking at diff against the upstream.

> Coverity should be smart enough to figure such things out and I am
> surprised it was not caught by our Coverity build.

Yeah, I would've thought so too. Does anyone pay attention at the 
Coverity results for things like this?

- Heikki

File: src/backend/access/transam/xlog.c (hunk 1 above)
Comment (hlinnaka) [inline src/backend/access/transam/xlog.c]: On 11/06/2015 08:26 PM, Asim RP wrote:
>> @@ -757,6 +759,7 @@ static char *str_time(pg_time_t tnow);
>>   #ifdef suppress
>>   static void issue_xlog_fsync(void);
>>   #endif
>> +static void xlog_outrec(StringInfo buf, XLogRecord *record);
>
> What's the convention we should follow?  Static functions should be named in lower case?

It's spelled in lower-case in upstream, that's the convention to follow. 
For GPDB-added functions, whatever seems to be cleanest in the context. 
There are many different styles in different parts of the codebase, so 
the best advice is to just follow the style of the surrounding code.

- Heikki

File: src/backend/access/transam/xlog.c [hunk 2]
```diff
@@ -1960,12 +1963,6 @@ AdvanceXLInsertBuffer(bool new_segment)
 	return update_needed;
 }
 
-void XLogGetBuffer(int startidx, int npages, char **from, Size *nbytes)
```
Comment (danielgustafsson) [inline src/backend/access/transam/xlog.c]: This is referenced in `src/test/unit/mock/mock_info.json` which in turn is supposed to be regenerated by `generate_mocks.py` which isn't in the main repo.

**Comment by danielgustafsson** [comment]
Not sure how to handle the `mock_info.json` reference (if at all), :+1: on all the rest

File: src/backend/access/transam/xlog.c (hunk 2 above)
Comment (hlinnaka) [inline src/backend/access/transam/xlog.c]: On 11/07/2015 12:24 AM, Daniel Gustafsson wrote:
>> @@ -1960,12 +1963,6 @@ AdvanceXLInsertBuffer(bool new_segment)
>>   	return update_needed;
>>   }
>>
>> -void XLogGetBuffer(int startidx, int npages, char **from, Size *nbytes)
>
> This is referenced in `src/test/unit/mock/mock_info.json` which in
> turn is supposed to be regenerated by `generate_mocks.py` which isn't
> in the main repo.

Hmm, I think generate_mocks.py was removed from the codebase years ago. 
It was part of some older mechanism to create the *_mock.c files.

Actually, looks like the whole mock_info.json file is unnecessary. I 
just tried replacing it with an empty json file (just {}), and ran "make 
unittest-check", and it worked great. All that dead stuff in the mock 
tests need to be cleaned up too...

- Heikki

Comment (danielgustafsson) [inline src/backend/access/transam/xlog.c]: Ok, makes sense, I didn't follow the rabbithole all the way.. another entry for the TODO.

**Comment by hlinnaka** [comment]
Pushed, thanks for the review @ashwinstar, @asimrp and @danielgustafsson!

**Pull request closed without merging**